rest/firebaseQueryHandler.go

```diff
@@ -38,6 +40,73 @@ func GetTracks(app *firebase.App, query models.FirebaseQuery) (data []models.Igc
 	return data, err
 }
 
+// GetTrack gets a track file from the Firebase Storage based on TrackID in metadata
+func GetTrack(app *firebase.App, trackID string) (data []byte, err error) {
+	client, err := app.Storage(context.Background())
+	if err != nil {
+		gtbackend.DebugLog(fileNameFUH, "GetTrack", err)
+		return
+	}
+
+	bucket, err := client.DefaultBucket()
+	if err != nil {
+		gtbackend.DebugLog(fileNameFUH, "GetTrack", err)
+		return
+	}
+
+	// Read the entire file to data
+	rc, err := bucket.Object(trackID).NewReader(context.Background())
+	data, err = ioutil.ReadAll(rc)
+	defer rc.Close()
+
+	return
+}
+
+// DeleteTrack deletes the track from storage and firestore
+func DeleteTrack(app *firebase.App, trackID string) (httpCode int, err error) {
+	ctx := context.Background()
+
+	// Delete file from storage
+	storageClient, err := app.Storage(ctx)
+	if err != nil {
+		gtbackend.DebugLog(fileNameFUH, "DeleteTrack - StorageClient", err)
+		httpCode = http.StatusBadRequest
+		return
+	}
```

Comment:
Could this whole if be extracted as a method seen as you do it 5 times in a row in this method alone?

Reply:
I moved the httpCode = http.StatusBadRequest up to the top of the function, return needs to be there and log needs to run, so extracting either is not possible, and moving them to an other function would still require the same arguments and one line of code per error test.

Reply:
I mean if it can't be refactored then it cannot be refactored

Reply:
I mean it is better, as long as only err is checked at return